src/lib.rs

```diff
@@ -439,10 +448,67 @@ macro_rules! thin_vec {
 }
 
 impl<T> ThinVec<T> {
+    /// Creates a new empty ThinVec.
+    ///
+    /// This will not allocate.
     pub fn new() -> ThinVec<T> {
         ThinVec::with_capacity(0)
     }
 
+    /// Constructs a new, empty `ThinVec<T>` with at least the specified capacity.
+    ///
+    /// The vector will be able to hold at least `capacity` elements without
+    /// reallocating. This method is allowed to allocate for more elements than
+    /// `capacity`. If `capacity` is 0, the vector will not allocate.
+    ///
+    /// It is important to note that although the returned vector has the
+    /// minimum *capacity* specified, the vector will have a zero *length*.
+    ///
+    /// If it is important to know the exact allocated capacity of a `ThinVec`,
+    /// always use the [`capacity`] method after construction.
+    ///
+    /// **NOTE**: unlike Vec, ThinVec **MUST** allocate to keep track of non-zero
```

Comment:
Nit: Vec and ThinVec here don't have backticks. I'm not sure how hard you're trying to get every one.